src/client/prerender.js

```diff
@@ -46,9 +46,17 @@ export function loadPrerenderScripts(initialState) {
 
 export function moveHelmetScripts() {
   document.addEventListener('DOMContentLoaded', () => {
-    const helmetScripts = [...document.querySelectorAll('script[data-react-helmet]')];
-    helmetScripts.forEach((script) => document.body.removeChild(script));
-    helmetScripts.forEach((script) => document.head.appendChild(script));
+    const headHelmetScripts = [...document.head.querySelectorAll('script[data-react-helmet]')];
+    const bodyHelmetScripts = [...document.body.querySelectorAll('script[data-react-helmet]')];
+
+    bodyHelmetScripts.forEach((script) => document.body.removeChild(script));
+
+    // If the number of helmet scripts in the head equals the number of helmet scripts
+    // the body, then we can assume that those scripts were already added by
+    // react-helmet because one-app would not have added them to the head yet
+    if (headHelmetScripts.length !== bodyHelmetScripts.length) {
```

Comment:
I think there are too many potential variables with this approach, its possible that more head helmet scripts are added than originally set in the body in the server render as you can't guarantee that the server bundle and client bundles won't always match up.

Reply:
I wonder about that. How would more head helmet scripts get added before this function is called? Wouldn't react-helmet (or equivalent) be the only other library responsible for inserting helmet scripts? Where would the extra helmet scripts come from? One-app wouldn't add a helmet script before this so it couldn't have done that. 

I can't think of a concrete example where that would happen so I would be interested to know what you're thinking of exactly.